temp_functions.py

```diff
@@ -0,0 +1,75 @@
+import read_db
+import pandas as pd
+import time
+from tqdm import tqdm
+
+class functions():
+
+    def __init__(self):
+        self.df=None
+        self.dict={}
+
+    def set_dataframe(self):
+        extract = read_db.ExtractData()
+        self.df = extract.get_timeslots_info()
+
+    def main(self):
+        self.set_dataframe()
+        temp = {}
+        for i,row in tqdm(self.df.iterrows()):
+            key = row["EmployeeID"], row["RelationID"]
+
+            ## Calculate the remaining availability in each calendar month and week
+            appointmentLength = (row["UntilUtc"] - row["FromUtc"]).total_seconds()/3600
+            if key[0] in temp:
+                if temp[key[0]]["DateLastWorked"].year == row["FromUtc"].year:
+                    if temp[key[0]]["DateLastWorked"].month == row["FromUtc"].month:
+                        temp[key[0]]["HoursLeftInMonth"] = temp[key[0]]["HoursLeftInMonth"] - appointmentLength
+                    else: 
+                        temp[key[0]]["HoursLeftInMonth"] = row["AverageNumberOfHoursPerMonth"] - appointmentLength
+                    if temp[key[0]]["DateLastWorked"].isocalendar().week == row["FromUtc"].isocalendar().week:
+                        temp[key[0]]["HoursLeftInWeek"] = temp[key[0]]["HoursLeftInWeek"] - appointmentLength
+                    else: 
+                        temp[key[0]]["HoursLeftInWeek"] = row["NumberOfHoursPerWeek"] - appointmentLength
+                else: 
+                    temp[key[0]]["HoursLeftInMonth"] = row["AverageNumberOfHoursPerMonth"] - appointmentLength
+                    temp[key[0]]["HoursLeftInWeek"] = row["NumberOfHoursPerWeek"] - appointmentLength
+                temp[key[0]]["DateLastWorked"] = row["UntilUtc"]
+            else:
+                temp[key[0]] = {"HoursLeftInMonth": row["AverageNumberOfHoursPerMonth"] - appointmentLength, 
+                                "HoursLeftInWeek": row["NumberOfHoursPerWeek"] - appointmentLength,
+                                "DateLastWorked": row["FromUtc"]}
+
+            ## Calculate the days since last visit and the total number of visits
+            if key in temp:
+                DaysSinceLastVisit = (row["UntilUtc"] - temp[key]["DateOfLastVisit"]).days
```

Comment:
Niet gebruikt

Reply:
row['fromutc']??

Reply:
> Niet gebruikt

Aan het einde van de function worden ze allemaal in een dict gezet om ze makkelijker uit de class te trekken.



> row['fromutc']??

We pakken gewoon de datum van de appointment, dus het maakt niet uit welke variable je pakt zolang het maar de datum van die appointment bevat.